src/test/scala/io/iohk/ethereum/transactions/PendingTransactionsManagerSpec.scala

```diff
@@ -155,6 +155,30 @@ class PendingTransactionsManagerSpec extends FlatSpec with Matchers with ScalaFu
       EtcPeerManagerActor.SendMessage(SignedTransactions(Seq(stx)), peer1.id))
   }
 
+  it should "remove transaction on timeout" in new TestSetup {
+    override val txPoolConfig = new TxPoolConfig {
+      override val txPoolSize: Int = 300
+      override val transactionTimeout: FiniteDuration = 1.seconds
+
+      //unused
+      override val pendingTxManagerQueryTimeout: FiniteDuration = Timeouts.veryLongTimeout
+    }
+
+    override val pendingTransactionsManager = system.actorOf(
+      PendingTransactionsManager.props(txPoolConfig, peerManager.ref, etcPeerManager.ref, peerMessageBus.ref))
+
+    val stx = newStx()
+    pendingTransactionsManager ! AddTransactions(stx)
+
+    val pendingTxs = (pendingTransactionsManager ? GetPendingTransactions).mapTo[PendingTransactionsResponse].futureValue
+    pendingTxs.pendingTransactions.map(_.stx).toSet shouldBe Set(stx)
+
+    Thread.sleep(1100)
```

Comment:
Is there a chance we can make this wait shorter? A second here, a second there...

Reply:
how about half a second?